test/e2e_env/multizone/meshservice/policies.go

```diff
@@ -225,4 +257,155 @@ spec:
 			g.Expect(response.Instance).To(HavePrefix("second-test-server"))
 		}, "30s", "1s").Should(Succeed())
 	})
+
+	It("should configure MeshRetry for MeshService in another zone", func() {
+		Expect(YamlK8s(fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshFaultInjection
+metadata:
+  name: mesh-fi-ms-another-zone
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+    kuma.io/origin: zone
+spec:
+  targetRef:
+    kind: Mesh
+    proxyTypes: ["Sidecar"]
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+          http:
+            - abort:
+                httpStatus: 503
+                percentage: "50"
+`, Config.KumaNamespace, meshName))(multizone.KubeZone2)).To(Succeed())
+		// given
+		// create a tunnel to test-client admin
+		Expect(multizone.KubeZone1.PortForwardService("test-client", namespace, 9901)).To(Succeed())
+		portFwd := multizone.KubeZone1.GetPortForward("test-client")
+		tnl := tunnel.NewK8sEnvoyAdminTunnel(multizone.Global.GetTesting(), portFwd.ApiServerEndpoint)
+
+		// then
+		Eventually(func(g Gomega) {
+			response, err := client.CollectFailure(
+				multizone.KubeZone1,
+				"test-client",
+				fmt.Sprintf("test-server.%s.svc.%s.mesh.local", namespace, Kuma2),
+				client.FromKubernetesPod(namespace, "test-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(response.ResponseCode).To(Equal(503))
+		}, "30s", "1s").Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			g.Expect(retryStat(tnl)).To(stats.BeEqualZero())
```

Comment:
`upstream_rq_retry_success` is a counter, it doesn't make sense to call inside `Eventually` it's either 0 from the beginning or it won't be 0 ever again